src/components/graph/analysis_window/header/Header.tsx

```diff
@@ -189,16 +191,28 @@ const LabelsAndTags: FC = () => {
           }}
         />
       ))}
-      <TagInput
-        currentAddressTags={addressCustomTags}
-        currentUserTags={userCustomTags}
-        onCreateCustomAddressTag={onCreateCustomAddressTag}
-        onDeleteCustomAddressTag={onDeleteCustomAddressTag}
-      />
-    </span>
+      <div onClick={() => {
+        if (!isAuthenticated) {
+          localStorage.setItem("focusedAddressData", JSON.stringify(analysisData));
```

Comment:
Good detail. I actually think you could just set the focused address data in the local storage whenever it changes and build this right into the useEffect rather than have it here. I don't think setting stuff in the local storage is particularly expensive and it'll greatly simplify the code.

Reply:
Great feedback. Also, it might be useful to have a global graph state, where the focused address data is included inside the graph info. This way, every time there is the need to persist the graph data (on page redirect for example)

Reply:
Great feedback. Also, it might be useful to have a global graph state, where the focused address data is included inside the graph info. This way, every time there is the need to persist the graph data (on page redirect for example)